module/x/icaauth/keeper/msg_server.go

```diff
@@ -0,0 +1,59 @@
+package keeper
+
+import (
+	"context"
+
+	"github.com/Gravity-Bridge/Gravity-Bridge/module/x/icaauth/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+)
+
+var _ types.MsgServer = msgServer{}
+
+type msgServer struct {
+	Keeper
+}
+
+// NewMsgServerImpl creates and returns a new types.MsgServer, fulfilling the icaauth Msg service interface
+func NewMsgServerImpl(keeper Keeper) types.MsgServer {
+	return &msgServer{Keeper: keeper}
+}
+
+// RegisterAccount implements the Msg/RegisterAccount interface
+func (k msgServer) RegisterAccount(goCtx context.Context, msg *types.MsgRegisterAccount) (*types.MsgRegisterAccountResponse, error) {
+	ctx := sdk.UnwrapSDKContext(goCtx)
+
+	acc, err := sdk.AccAddressFromBech32(msg.Owner)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := k.RegisterInterchainAccount(ctx, acc, msg.ConnectionId, msg.Version); err != nil {
+		return nil, err
+	}
+
+	return &types.MsgRegisterAccountResponse{}, nil
+}
+
+// SubmitTx implements the Msg/SubmitTx interface
+func (k msgServer) SubmitTx(goCtx context.Context, msg *types.MsgSubmitTx) (*types.MsgSubmitTxResponse, error) {
```

Comment:
MsgSubmitTx will contain an array of Msgs which should be submitted to the foreign chain's ICA host module and executed as the interchain account.